packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol

```diff
@@ -404,7 +404,9 @@ contract SuperchainWETH_Test is CommonTest {
         superchainWeth.transferFrom(_src, _dst, _wad);
 
         // Assert
-        assertEq(superchainWeth.balanceOf(_src), 0);
         assertEq(superchainWeth.balanceOf(_dst), _wad);
+        // Handle the case where the source and destination are the same to check the source balance.
+        if (_src != _dst) assertEq(superchainWeth.balanceOf(_src), 0);
+        else assertEq(superchainWeth.balanceOf(_src), _wad);
```

Comment:
isnt it cleaner to assume `src != dst` ?

I know we are ignoring that explicit case but idk if it matters, no strong opinion

Reply:
To be consistent with how are we doing, would it be better to test that scenario on another test? I think it's valuable to cover that case as well

Reply:
yeah better another test!